llama_index/evaluation/correctness.py

```diff
@@ -125,9 +125,8 @@ async def aevaluate(
 
         await asyncio.sleep(sleep_time_in_seconds)
 
-        if query is None or response is None or reference is None:
-            print(query, response, reference, flush=True)
-            raise ValueError("query, response, and reference must be provided")
+        if query is None or response is None:
```

Comment:
Since `reference` is used in the predict below, shouldn't it also be checked for `None` ?

Reply:
So just confirming, this will leave a blank space in the prompt template, and that's ok?

```
DEFAULT_USER_TEMPLATE = """
## User Query
{query}

## Reference Answer
{reference_answer}

## Generated Answer
{generated_answer}
"""
```

Reply:
Okay so it does work when we pass in a None for reference answer and also the scoring rubric doesn't really say to use the reference answer explicitly, which is why I think this works when it is None.

But to be more safe, I modified the prompt to say that "you may also be given a reference answer". And when, it is None we pass "(NO REFERENCE ANSWER SUPPLIED)" string

Reply:
Just did some testing on this with GPT-3.5 using MT-bench single-grading that doesn't have reference_answer; and the feedback/scores look good to me. I think this should be okay.